### PR TITLE
AKS - baseline - Convert Bootstrap prep deployment file to bicep (acr-stamp)

### PR DESCRIPTION
**Pull request by magrande**

### CODE REVIEW

File: acr-stamp.bicep [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+    retentionInDays: 30
+  }
+}
+
+// Limit access to a the ACR by assigning virtual network private IP addresses to the registry endpoints and using Azure Private Link
```
Comment (ckittel) [inline acr-stamp.bicep:88]:
```suggestion
// Azure Container Registry will be exposed via Private Link, set up the related Private DNS zone and virtual network link to the spoke.
```

File: acr-stamp.bicep [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  name: '${last(split(targetVnetResourceId,'/'))}'
+  
+  resource snetClusterNodes 'subnets@2021-05-01' existing = {
+    name: 'snet-clusternodes'
+  }
+}
+
+/*** RESOURCES ***/
+
+resource laAks 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
```
Comment (ckittel) [inline acr-stamp.bicep:77]:
```suggestion
// This Log Analytics workspace will be the log sink for all resources in the cluster resource group. This includes ACR, the AKS cluster, Key Vault, etc. It also is the Container Insights log sink for the AKS cluster.
resource laAks 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
```

File: acr-stamp.bicep [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    properties: {
+      virtualNetwork: {
+        id: targetVnetResourceId
+      }
+      registrationEnabled: false
+    }
+  }
+}
+
+// The Container Registry that will be used when bootstrapped the AKS cluster
```
Comment (ckittel) [inline acr-stamp.bicep:106]:
```suggestion
// The Container Registry that the AKS cluster will be authorized to use to pull images.
```

File: acr-stamp.bicep [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    ]
+    logs: [
+      {
+        category: 'ContainerRegistryRepositoryEvents'
+        enabled: true
+      }
+      {
+        category: 'ContainerRegistryLoginEvents'
+        enabled: true
+      }
```
Comment (ckittel) [inline acr-stamp.bicep:168]: Can you convert this to the `allLogs` category group?

File: acr-stamp.bicep [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+          privateLinkServiceId: acrAks.id
+          groupIds: [
+            'registry'
+          ]
+        }
+      }
+    ]
+  }
+
+  // Since the private endpoint is integrated with a private DNS zone, a private DNS zone group is also created.
```
Comment (ckittel) [inline acr-stamp.bicep:197]:
```suggestion
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Looks pretty good!  Couple suggestions below.

File: acr-stamp.bicep (hunk 4 above)
Comment (magrande) [reply acr-stamp.bicep]: Done

File: acr-stamp.bicep [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        status: 'enabled'
+      }
+    }
+    publicNetworkAccess: 'Disabled'
+    encryption: {
+      status: 'disabled'
+    }
+    dataEndpointEnabled: true
+    networkRuleBypassOptions: 'AzureServices'
+    zoneRedundancy: 'Disabled'
```
[the remark this answers is not in the document]
Comment (magrande) [reply acr-stamp.bicep]: put it back. Good catch

File: acr-stamp.bicep [hunk 7]
```diff
@@ -0,0 +1,215 @@
+targetScope = 'resourceGroup'
+
+/*** PARAMETERS ***/
+
+@description('The regional network spoke VNet Resource ID that the cluster will be joined to.')
+@minLength(1)
```
[the remark this answers is not in the document]
Comment (magrande) [reply acr-stamp.bicep]: Sorry for that, I saw it but skipped it...

File: acr-stamp.bicep (hunk 4 above)
Comment (ckittel) [reply acr-stamp.bicep]: Thanks.

File: acr-stamp.bicep (hunk 7 above)
[the remark this answers is not in the document]
Comment (ckittel) [reply acr-stamp.bicep]: Thanks for updating it.

File: acr-stamp.bicep (hunk 6 above)
[the remark this answers is not in the document]
Comment (ckittel) [reply acr-stamp.bicep]: w00t.

**Comment by ckittel** [comment]
Thank you @magrande and @teilmeier (ref: #285) for this next step in the migration.  Once all the templates have been migrated in-place, we'll be coming back to building out the module per-resource direction that @teilmeier is proposing in #286. We might even go down a private bicep registry part for the last step (since we have an ACR instance separate from the last step).  All TBD and is on the backlog for consideration.

**Pull request merged**